web3/_utils/abi.py

```diff
@@ -611,17 +611,21 @@ def normalizer(datatype, data):
     2. Recursively mapping each of the normalizers to the data
     3. Stripping the types back out of the tree
     """
-    pipeline = itertools.chain(
-        [abi_data_tree(types)],
-        map(data_tree_map, normalizers),
-        [partial(recursive_map, strip_abi_type)],
+    return pipe(
+        data, 
+        # 1. Decorating the data tree with types
+        abi_data_tree(types), 
+        # 2. Recursively mapping each of the normalizers to the data
+        IteratorProxy,
+        *map(data_tree_map, normalizers), 
+        # 3. Stripping the types back out of the tree
+        strip_abi_types,
+        list,
     )
 
-    return pipe(data, *pipeline)
-
 
 @curry
-def abi_data_tree(types: Sequence[TypeStr], data: Sequence[Any]) -> List[Any]:
+def abi_data_tree(types: Sequence[TypeStr], data: Sequence[Any]) -> "map[ABITypedData]":
```

Comment:
idea for more optimization: rip out this func and replace it with a lambda which would have less overhead than using curry